tests/queries/0_stateless/02834_add_date_function.sql

```diff
@@ -0,0 +1,9 @@
+SELECT addDate('2022-05-07'::Date, INTERVAL 5 MINUTE);
+SELECT addDate('2022-05-07'::Date32, INTERVAL 5 MINUTE);
+SELECT addDate('2022-05-07'::DateTime, INTERVAL 5 MINUTE);
+SELECT addDate('2022-05-07'::DateTime64, INTERVAL 5 MINUTE);
```

Comment:
Maybe test what happens if addition moves the result beyond the bounds of the date time type?

Reply:
The behavior will be the same as in the case with the `+` or `-` operations: `1900-01-01 00:00:00.000` with the required precision.

In MySQL we simply get NULL. Should we try to replicate this behavior?

Reply:
So, if the result is outside the bounds of the data type, we say that the result is undefined.
The same semantics is used in `makeDate()`: https://fiddle.clickhouse.com/e58005b1-95f7-49fd-9989-10e295bfb82f

In my view, there is no need to replicate MySQL's behavior, it is more important to stay consistent within ClickHouse's functions and it's an edge case anyways.